clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,239 @@
+use std::{
+    env, fs,
+    io::{self, Read},
+    process::{Command, Stdio},
+};
+
+use serde_json::json;
+
+use crate::config::TeliodDaemonConfig;
+
+struct PathConfig;
+
+impl PathConfig {
+    const APP_TMP_DIR: &'static str = "/tmp/nordsecuritymeshnet";
+    const QPKG_DIR: &'static str = "/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet";
+
+    pub fn teliod_bin() -> String {
+        format!("{}/teliod", Self::QPKG_DIR)
+    }
+
+    pub fn teliod_cfg() -> String {
+        format!("{}/teliod.cfg", Self::QPKG_DIR)
```

Comment:
Teliod knowing where is the config file is a bit weird idea imho, in such case I would refactor it to exclude that argument and hardcode it in the rest of the code as well.

Reply:
sorry I didn't understand, can you explain?

Reply:
Currently teliod daemon starts with config file path provided as an argument - the situation where this path is at the same time hardcoded provided to and hardcoded in teliod doesn't feel right.

Reply:
Other platforms certainly have a different path for the config file whereas for QNAP this is a static path, hence the idea of having it hardcoded.